### DISCUSSION TITLE
#Validate and rules() function do not merge rules validations

### DISCUSSION THREAD

**Discussion by windsondias** [livewire/livewire#7360] [7. Bugs (archived)]
### Livewire version

3.2.1

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

When I use #Validate at the same time as the rule() function, the function's rules are ignored:

    #[Validate(['required'])]
    public ?string $username;

    protected function rules(): array
    {
        return [
            'username' => Rule::unique(User::class)->ignore($this->user())
        ];
    }

In this case I want to use #Validate for the facilities it offers, but I need to use the rules() function to check the database.

To make the example easier, replace the Rule class with a basic rule

    protected function rules(): array
    {
        return [
            'username' => 'max:5'
        ];
    }

Now when I fill in fields with more than 5 characters the form submits without validating

![image](https://github.com/livewire/livewire/assets/57019436/aef9e887-0c89-4d51-91f7-009af634f7fa)

Now, if I comment out the #Validate line and submit again, the function rule is applied

    //  #[Validate(['required'])]
    public ?string $username;

![image](https://github.com/livewire/livewire/assets/57019436/964df207-cda8-40b4-83bc-116a8181e7ce)

This already happened with #Rule, forcing the rule() function to be used in a single field for all validations. After the change in PR #7260  I thought it might be resolved but it still persists

The example I practically gave is cited in the PR itself. I believe it is a bug, but it could be something specific to my environment

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by BoGnY** [comment]
You cannot use both together...
See warning on [Livewire validation rules()](https://livewire.laravel.com/docs/validation#defining-a-rules-method).

This work:
```php
public ?string $username;

protected function rules(): array
{
    return [
        'username' => ['required', Rule::unique(User::class)->ignore($this->user())]
    ];
}

```
